provider/aws.go

```diff
@@ -254,6 +257,58 @@ func (p *AWSProvider) submitChanges(changes []*route53.Change) error {
 	return nil
 }
 
+func limitChangeSet(cs []*route53.Change, limit int) []*route53.Change {
+	if len(cs) <= limit {
+		return cs
+	}
+
+	log.Warningf("Initial change batch count is %d", len(cs))
+
+	changesByName := make(map[string][]*route53.Change, 0)
+	for _, v := range cs {
+		changesByName[*v.ResourceRecordSet.Name] = append(changesByName[*v.ResourceRecordSet.Name], v)
+	}
+
+	names := make([]string, 0)
+	for v := range changesByName {
+		names = append(names, v)
+	}
+	sort.Strings(names)
+
+	limCs := make([]*route53.Change, 0)
+	for i := 0; i < len(names); i++ {
+		changes := changesByName[names[i]]
+		if (limit - len(limCs)) >= len(changes) {
+			limCs = append(limCs, changes...)
+		}
+	}
+	limCs = sortChangesByActionNameType(limCs)
+
+	log.Warningf("Limited change batch count to %d", len(limCs))
+
+	return limCs
+}
+
+func sortChangesByActionNameType(cs []*route53.Change) []*route53.Change {
+	sort.SliceStable(cs, func(i, j int) bool {
```

Comment:
with this change travis builds will break on go1.7, but I think we can safely drop it from travis. Would want to do it, or u want me to update it on this PR ?